src/llmcompressor/modifiers/awq/base.py

```diff
@@ -96,53 +116,147 @@ class AWQModifier(Modifier):
     # Allow arbitrary types because AWQMapping has fields of type torch.nn.Module
     model_config: ConfigDict = ConfigDict(arbitrary_types_allowed=True)
 
+    # User-provided vars (in addition to QuantizationMixin args)
+    sequential_targets: Union[str, List[str], None] = None
     mappings: List[AWQMapping] = AWQ_MAPPING_REGISTRY["Llama"]
-    ignore: List[str] = []
-    group_size: int = 128
     max_chunk_memory: int = 1024 * 1024 * 1024
-    num_bits: int = 4
-    symmetric: bool = False
     duo_scaling: bool = True
 
-    _resolved_mappings: List[ResolvedMapping] = []
-    _scales: Dict[str, Union[torch.Tensor, List[torch.Tensor]]] = {}
-    _module_kwargs: Dict = {}
+    # Private vars set during validation
+    _num_bits: Optional[int] = PrivateAttr(default=None)
+    _symmetric: Optional[bool] = PrivateAttr(default=None)
+    _group_size: Optional[int] = PrivateAttr(default=None)
+
+    # Private vars set during initialization, cleared during finalization
+    _resolved_mappings: List[ResolvedMapping] = PrivateAttr(default_factory=list)
+    _activations: Dict[str, List[torch.Tensor]] = PrivateAttr(default_factory=dict)
+    _activation_hooks: Set[RemovableHandle] = PrivateAttr(default_factory=set)
+    _module_kwargs: Dict = PrivateAttr(default_factory=dict)
+
+    @model_validator(mode="after")
+    def validate_model_after(model: "AWQModifier") -> "AWQModifier":
+        if not model.config_groups and not model.scheme:
+            raise ValueError("AWQ requires either a config_groups or a scheme")
+
+        # TODO better way to do this?
```

Comment:
You can use `config = self.resolve_config()` and then validate that

Reply:
thanks!

Reply:
hmm, got a link? I don't see any `resolve_config` defined anywhere in the code

Reply:
found `.resolve_quantization_config`, thanks!